frontend/app/src/store/balances/non-fungible.ts

```diff
@@ -1,75 +1,156 @@
-import { NonFungibleBalance, NonFungibleBalances } from '@/types/nfbalances';
+import {
+  NonFungibleBalance,
+  NonFungibleBalancesCollectionResponse,
+  NonFungibleBalancesRequestPayload
+} from '@/types/nfbalances';
 import { Section, Status } from '@/types/status';
 import { TaskMeta } from '@/types/task';
 import { TaskType } from '@/types/task-type';
+import { Zero } from '@/utils/bignumbers';
+import {
+  defaultCollectionState,
+  mapCollectionResponse
+} from '@/utils/collection';
 import { logger } from '@/utils/logging';
 
 export const useNonFungibleBalancesStore = defineStore(
   'balances/non-fungible',
   () => {
-    const balances: Ref<NonFungibleBalances> = ref({});
+    const { itemsPerPage } = storeToRefs(useFrontendSettingsStore());
+
+    const defaultRequestPayload =
+      (): Partial<NonFungibleBalancesRequestPayload> => {
+        return {
+          offset: 0,
+          limit: get(itemsPerPage),
+          ignoredAssetsHandling: 'exclude'
+        };
+      };
+
+    const requestPayload: Ref<Partial<NonFungibleBalancesRequestPayload>> = ref(
+      defaultRequestPayload()
+    );
+
+    const balances = ref(defaultCollectionState<NonFungibleBalance>()) as Ref<
+      Collection<NonFungibleBalance>
+    >;
+
+    const nonFungibleTotalValue = ref<BigNumber>(Zero);
 
     const { activeModules } = storeToRefs(useGeneralSettingsStore());
-    const { awaitTask } = useTasks();
+    const { awaitTask, isTaskRunning } = useTasks();
     const { notify } = useNotifications();
     const { tc } = useI18n();
 
-    const nonFungibleBalances: ComputedRef<NonFungibleBalance[]> = computed(
-      () => {
-        const data: NonFungibleBalance[] = [];
-        const nonFungibleBalances = get(balances) as NonFungibleBalances;
-        for (const address in nonFungibleBalances) {
-          const addressNfBalance = nonFungibleBalances[address];
-          data.push(...addressNfBalance);
-        }
-        return data;
+    const updateRequestPayload = async (
+      payload: Partial<NonFungibleBalancesRequestPayload>
+    ) => {
+      if (!isEqual(payload, get(requestPayload))) {
+        set(requestPayload, payload);
+        await fetchNonFungibleBalances();
       }
-    );
+    };
 
-    const nonFungibleTotalValue = (
-      includeLPToken = false
-    ): ComputedRef<BigNumber> =>
-      computed(() => {
-        return bigNumberSum(
-          get(nonFungibleBalances)
-            .filter(item => includeLPToken || !item.isLp)
-            .map(item => item.usdPrice)
-        );
-      });
+    const { fetchNfBalances, fetchNfBalancesTask } = useNftBalanceApi();
+
+    const fetchNfTotalValue = async (): Promise<BigNumber> => {
+      const payload: NonFungibleBalancesRequestPayload = {
+        limit: 0,
+        offset: 0,
+        ignoreCache: false
+      };
+
+      const result = await fetchNfBalances(payload);
+      return mapCollectionResponse(result).totalUsdValue || Zero;
+    };
```

Comment:
For the total value, I send it without any filter (for the overall balances), meanwhile for the balances it may contain filter.

But I think I shouldn't call the total value here🧐

Reply:
I move the call for `fetchNfTotalValue`. I call it only on first load, or if the the fetchNfBalances is called with ignoreCache=false.

Is it more makes sense?

Reply:
Oh, I forgot about the dashboard overall completely. So we have two different usages for total.

One is the table total, which should be the filtered total USD value,
and the other is the one we sum to the dashboard which needs to be unfiltered.

Instead of doing a second call, can't we just update it (also add a comment there to not forget) whenever we call the endpoint without a filter?

Would that be too tricky to implement?

Reply:
updated